src/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -210,6 +210,12 @@ internal static Assembly LoadGeneratedAssembly(Type type, string defaultNamespac
 
                     return null;
                 }
+
+                if(!IsSerializerVersionMatch(serializer, type, defaultNamespace))
+                {
+                    throw new Exception(SR.Format(SR.XmlSerializerExpiredDetails, serializerName, type.FullName));
```

Comment:
When !IsSerializerVersionMatch is true, should we return null? Also why we throw exception here? I would expect that we return null to indicate no pre-generated assembly, then we will generate new pre-generated assembly and the program keeps going. Looks like the throwing exception is different behavior from desktop. However, I noticed the other places without matching pre-generated assembly throw as well in the same function . I'm wondering why.

Reply:
please put the code in the #if !FEATURE_SERIALIZATION_UAPAOT